dev/dev-env-setup.sh

```diff
@@ -145,189 +148,215 @@ version_gt() {
+  pyenv exec pip install $(quiet_command) virtualenv
+  pyenv exec pip install $(quiet_command) pre-commit
+}
 
 # Check if the virtualenv already exists at the specified path
-if [[ -d "$directory"  ]]; then
-  if [ -z "$GITHUB_ACTIONS" ]; then
-    read -p "A virtual environment is already located at $directory. Do you wish to replace it? $(tput bold; tput setaf 2)(y/n) $(tput sgr0)" -n 1 -r
-    echo
-  fi
-  if [[ $REPLY =~ ^[Yy]$ || -n "$GITHUB_ACTIONS" ]]; then
-    echo "Replacing Virtual environment in '$directory'. Installing new instance."
-    pyenv exec virtualenv --clear "$directory"
+create_virtualenv() {
+  if [[ -d "$directory" ]]; then
+    if [ -z "$GITHUB_ACTIONS" ]; then
+      read -p "A virtual environment is already located at $directory. Do you wish to replace it? $(
+        tput bold
+        tput setaf 2
+      )(y/n) $(tput sgr0)" -n 1 -r
+      echo
+    fi
+    if [[ $REPLY =~ ^[Yy]$ || -n "$GITHUB_ACTIONS" ]]; then
+      echo "Replacing Virtual environment in '$directory'. Installing new instance."
+      pyenv exec virtualenv --clear "$directory"
+    fi
+  else
+    # Create a virtual environment with the specified Python version
+    pyenv exec virtualenv --python "$PY_INSTALL_VERSION" "$directory"
   fi
-else
-  # Create a virtual environment with the specified Python version
-  pyenv exec virtualenv --python "$PY_INSTALL_VERSION" "$directory"
-fi
 
-# Activate the virtual environment
-# shellcheck disable=SC1090
-source "$VENV_DIR"
+  # Activate the virtual environment
+  # shellcheck disable=SC1090
+  source "$VENV_DIR"
 
-echo "$(tput setaf 2)Current Python version: $(tput bold)$(python --version)$(tput sgr0)"
-echo "$(tput setaf 3)Activated environment is located: $(tput bold) $directory/bin/activate$(tput sgr0)"
-
-echo "Installing pip dependencies for development environment."
+  echo "$(tput setaf 2)Current Python version: $(tput bold)$(python --version)$(tput sgr0)"
+  echo "$(tput setaf 3)Activated environment is located: $(tput bold) $directory/bin/activate$(tput sgr0)"
+}
 
 # Install current checked out version of MLflow (local)
-pip install $(quiet_command) -e .[extras]
-pip install pre-commit
-
-if [[ -n "$full" ]]; then
-  # Install dev requirements and test plugin
-  pip install $(quiet_command) -r "$rd/dev-requirements.txt"
-  # Install test plugin
-  pip install $(quiet_command) -e "$MLFLOW_HOME/tests/resources//mlflow-test-plugin"
-  echo "Finished installing pip dependencies."
-else
-  files=("$rd/test-requirements.txt" "$rd/lint-requirements.txt" "$rd/doc-requirements.txt")
-  for r in "${files[@]}";
-  do
-    pip install $(quiet_command) -r "$r"
-  done
-fi
-
-echo "$(tput setaf 2; tput smul)Python packages that have been installed:$(tput rmul)"
-echo "$(pip freeze)$(tput sgr0)"
+install_mlflow() {
+  echo "Installing pip dependencies for development environment."
+
+  pip install $(quiet_command) -e .[extras]
+  pip install pre-commit
```

Comment:
let's remove `pre-commit`. It's in `lint-requirements.txt`.

Reply:
got it